gui.py

```
@@ -366,16 +271,28 @@ def run(screen: pygame.Surface = None):
                 mousbuttondown = False
             elif event.type == pygame.KEYDOWN:
-                zlayer[0].handle_input(event)
-
+                if event.key == pygame.K_F12 or event.key == 1073742051:
+                    if menu_opend:
+                        menu_opend = False
+                    else:
+                        menu_opend = True
+                else:
+                    if zlayer.count() > 0 and selected_window:
+                        if type(zlayer[0]) == MosWindow or type(zlayer[0]) == MosTerminal:
```

Ruff (0.13.1)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)